presto-hive/src/main/java/io/prestosql/plugin/hive/HiveType.java

```diff
@@ -93,11 +93,19 @@ public TypeInfo getTypeInfo()
         return typeInfo;
     }
 
+    /** @deprecated Prefer {@link #getTypeSignature(HiveTimestampPrecision)}. */
+    @Deprecated
     public TypeSignature getTypeSignature()
     {
-        return toTypeSignature(typeInfo);
+        return getTypeSignature(DEFAULT_PRECISION);
```

Comment:
I think renaming the method to something like `getTypeWithUnknownTimePrecision` might be a better idea if we don't plan to fix it, because it makes the meaning more clear.

If `TypeManager` were local to Hive, it would be a better idea to include the timestamp precision there, which could eliminate all but one use of `getTypeSignature()`.

Reply:
> I think renaming the method to something like getTypeWithUnknownTimePrecision might be a better idea if we don't plan to fix it, because it makes the meaning more clear.

Sounds good!
(nit: TimePrecision -> TimestampPrecision)

Reply:
let's address this as a followup